xbmc/interfaces/json-rpc/AudioLibrary.cpp

```diff
@@ -580,19 +580,35 @@ JSONRPC_STATUS CAudioLibrary::SetAlbumDetails(const std::string &method, ITransp
     return InternalError;
 
   CAlbum album;
-  if (!musicdatabase.GetAlbum(id, album) || album.idAlbum <= 0)
+  // Get current album details, but not songs as we do not want to update them here
+  if (!musicdatabase.GetAlbum(id, album, false) || album.idAlbum <= 0)
     return InvalidParams;
 
   if (ParameterNotNull(parameterObject, "title"))
     album.strAlbum = parameterObject["title"].asString();
-  // Artist names, along with MusicbrainzAlbumArtistID needs to update artist credits
-  // As temp fix just set the album artist description string
+  if (ParameterNotNull(parameterObject, "displayartist")) 
+    album.strArtistDesc = parameterObject["displayartist"].asString();
+  // Set album sort string before processing artist credits
+  if (ParameterNotNull(parameterObject, "sortartist"))
+    album.strArtistSort = parameterObject["sortartist"].asString();
+
+  // Match up artist names and mbids to make new artist credits
+  // Mbid values only apply if there are names
   if (ParameterNotNull(parameterObject, "artist"))
   {
-    std::vector<std::string> artist;
-    CopyStringArray(parameterObject["artist"], artist);
-    album.strArtistDesc = StringUtils::Join(artist, g_advancedSettings.m_musicItemSeparator);
+    std::vector<std::string> artists, mbids;
```

Comment:
Could you explicitly declare both variables on separate lines? I always find that declaring multiple variables on the same line results in people missing all but the first (or the last) variable name.